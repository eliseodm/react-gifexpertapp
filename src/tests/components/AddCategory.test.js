import React from 'react';
import '@testing-library/jest-dom';
const { shallow } = require("enzyme")
const { AddCategory } = require("../../components/AddCategory")


describe('Pruebas en el componente <AddCategory />', () => {

    const setCategories =  jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } />);

    beforeEach(() => {
        
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } />);

    })

    test('Deberia de mostrar correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('debe de buscar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe( value );

    })

    test('No debe de postear la informacion onSubmit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
        
    })
    
    
    
})
