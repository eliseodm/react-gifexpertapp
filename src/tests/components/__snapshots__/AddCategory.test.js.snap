// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Deberia de mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
